driver-core/src/main/com/mongodb/operation/OperationHelper.java

```diff
@@ -77,6 +79,30 @@ public Void apply(final T t) {
         }
     }
 
+    static void checkValidReadConcern(final Connection connection, final ReadConcern readConcern) {
+        if (!serverIsAtLeastVersionThreeDotTwo(connection.getDescription()) && !readConcern.equals(ReadConcern.DEFAULT)) {
```

Comment:
For all of these checks: can readConcern ever be null?  I think we should assume non-null and force operations to check for null.